src/app/components/file-upload/file-upload.component.ts

```diff
@@ -1,40 +1,173 @@
 /* eslint-disable no-console */
 /* eslint-disable no-undef */
-import { Component, OnInit } from '@angular/core';
+import {
+  Component, OnInit, ViewChild, ElementRef
+} from '@angular/core';
 import { FormBuilder, FormGroup } from '@angular/forms';
 import { HttpClient } from '@angular/common/http';
+import { Observable } from 'rxjs';
+import yaml from 'js-yaml';
+
+// NgRX
+import { Store } from '@ngrx/store';
+import { Router } from '@angular/router';
+import { SwagPaths } from '../../state/swag-paths/swag-paths.model';
 
 @Component({
   selector: 'app-file-upload',
   templateUrl: './file-upload.component.html',
   styleUrls: ['./file-upload.component.scss'],
 })
 export class FileUploadComponent implements OnInit {
-  private baseUrl = 'http://localhost:8083/Docutest';
+  // NgRX
+  swagPaths$: Observable<SwagPaths>;
+
+  private Base_Url = 'http://localhost:8083/Docutest';
 
   public uploadForm: FormGroup;
 
-  constructor(private formBuilder: FormBuilder, private http: HttpClient) {}
+  public selectedFile: File;
+
+  public errorMsg: String = '';
+
+  public show = false;
+
+  public fileExt: string;
+
+  public regex = /\.([0-9a-z]+)(?=[?#])|(\.)(?:[\w]+)$/gmi;
+
+  /* Used to reset the file input */
+  @ViewChild('fileIn')
+  fileInput: ElementRef;
+
+  public swag: any;
+
+  public paths: Array<any>;
+
+  constructor(private formBuilder: FormBuilder, private http: HttpClient,
+    private store: Store<{swagPaths}>, public router: Router) {
+    this.swagPaths$ = store.select('swagPaths');
+  }
 
   ngOnInit(): void {
     this.uploadForm = this.formBuilder.group({
       swaggerFile: [''],
     });
   }
 
+  getFileExtension(file): string {
```

Comment:
This `file` parameter should have a type, if possible. Ideally it should be a specific type, not `any`.

Reply:
The same applies to the `swag` parameter of the methods below.